### PR TITLE
Make USM parameter bounds checking configurable

### PR DESCRIPTION
**Pull request by kbenzie**
Bounds checking can be an expensive operation to perform, especially for applications with heavy use of commands operating on USM allocations.

This patch adds a new layer specifically for USM bounds checking called UR_LAYER_BOUNDS_CHECKING. This was previously part of UR_LAYER_PARAMETER_VALIDATION.

By removing USM bounds checking from the default parameter validation the overhead of using parameter validation greatly decreases allowing it to be enabled in more situations.

### CODE REVIEW

**Review by pbalcer — APPROVED** [review]
I've added some info about how to update the baseline: https://github.com/oneapi-src/unified-runtime/pull/1954

tl;dr: after this is merged, please run the compute benchmarks job with PR 0, and parameter `--save baseline`.

**Comment by kbenzie** [comment]
> I've added some info about how to update the baseline: #1954
> 
> tl;dr: after this is merged, please run the compute benchmarks job with PR 0, and parameter `--save baseline`.

Done: https://github.com/oneapi-src/unified-runtime/actions/runs/10370029936
